/*
===========================================================================
*
*                            PUBLIC DOMAIN NOTICE                          
*               National Center for Biotechnology Information
*         Lister Hill National Center for Biomedical Communications
*                                                                          
*  This software is a "United States Government Work" under the terms of the
*  United States Copyright Act.  It was written as part of the authors' official
*  duties as a United States Government contractor and thus cannot be
*  copyrighted.  This software is freely available to the public for use. The
*  National Library of Medicine and the U.S. Government have not placed any
*  restriction on its use or reproduction.  
*                                                                          
*  Although all reasonable efforts have been taken to ensure the accuracy  
*  and reliability of the software and data, the NLM and the U.S.          
*  Government do not and cannot warrant the performance or results that    
*  may be obtained by using this software or data. The NLM and the U.S.    
*  Government disclaim all warranties, express or implied, including       
*  warranties of performance, merchantability or fitness for any particular
*  purpose.                                                                
*                                                                          
*  Please cite the authors in any work or product based on this material.   
*
===========================================================================
*/

/**
 * Example program for submitting an Interactive SemRep request.
 *
 * This example shows how to setup a basic Interactive SemRep request.
 * This runs the latest version of SemRep with Full Fielded Output (-D).
 * 
 * @author	Jim Mork
 * @version	1.0, June 16, 2011
**/
package test;

import java.io.BufferedWriter;
import java.io.File;
import java.io.FileWriter;
import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Path;
import java.nio.file.Paths;

public class NonAsciiString {

	public static void main(String a[]) throws IOException {
		String dir = "C:\\Users\\kelu\\Downloads\\public_mm\\";
		Path path = Paths.get(dir + "allsummarys.txt");
		String str = new String(Files.readAllBytes(path));
		
		// String str = "Bj��rk����oacute�";
		// str +=
		// "The MAQC-II Project: A comprehensive study of common practices for the development and validation of microarray-based predictive models The second phase of the MicroArray Quality Control (MAQC-II) project evaluated common practices for developing and validating microarray-based models aimed at predicting toxicological and clinical endpoints. The purposes of the MAQC-II project were to survey approaches in genomic model development in an attempt to understand sources of variability in prediction performance, and to assess the influences of endpoint signal strength in data. Thirty-six teams developed classifiers for 13 diverse endpoints -- some easy, some difficult to predict, from six relatively large training data sets -- three preclinical (toxicogenomics) and three clinical. By providing the same data sets to many organizations for analysis, but not restricting their data analysis protocols (DAPs), the project made it possible to evaluate to what extent, if any, results depend on the team that performs the analysis. These analyses collectively produced >18,000 models that were challenged by independent and blinded validation sets generated for MAQC-II. The cross-validated performance estimates for models developed under good practices are predictive of the blinded validation performance. The achievable prediction performance is largely determined by the intrinsic predictability of the endpoint, and simple data analysis methods often perform as well as more complicated approaches. Multiple models of comparable performance can be developed for a given endpoint and the stability of gene lists correlates with endpoint predictability. Importantly, similar conclusions were reached when >12,000 new models were generated by swapping the original training and validation sets. Description of six data sets including 13 prediction endpoints: (Summarized in GSE16716_MAQC-II_Datasets_Overview.pdf attached as supplementary file. For more details, see the MAQC-II main paper and its references for individual dataset.) The MAQC-II predictive modeling was limited to binary classification problems; therefore, continuous endpoint values such as overall survival (OS) and event-free survival (EFS) times were dichotomized using a milestone cutoff of censor data. Prediction endpoints were chosen to span a wide range of prediction difficulty. Two endpoints, H (CPS1) and L (NEP_S), representing the gender of the patients, were used as positive control endpoints, since they are easily predictable by microarrays. Two other endpoints, I (CPS1) and M (NEP_R), representing randomly assigned class labels, were designed to serve as negative control endpoints, since they are not supposed to be predictable. Data analysis teams were not aware of the characteristics of endpoints H, I, L, and M until their swap prediction results had been submitted. If a data analysis protocol did not yield models to accurately predict endpoints H and L, or if a data analysis protocol claims to be able to yield models to accurately predict endpoints I and M, something must have gone wrong. The Hamner data set (endpoint A) was provided by The Hamner Institutes for Health Sciences (Research Triangle Park, NC, USA). The study objective was to apply microarray gene expression data from the lung of female B6C3F1 mice exposed to a 13-week treatment of chemicals to predict increased lung tumor incidence in the 2-year rodent cancer bioassays of the National Toxicology Program. If successful, the results may form the basis of a more efficient and economical approach for evaluating the carcinogenic activity of chemicals. Microarray analysis was performed using Affymetrix Mouse Genome 430 2.0 arrays on three to four mice per treatment group, and a total of 70 mice were analyzed and used as the MAQC-II's training set. Additional data from another set of 88 mice were collected later and provided as the MAQC-II's external validation set. The Iconix data set (endpoint B) was provided by Iconix Biosciences, Inc. (Mountain View, CA, USA). The study objective was to assess, upon short term exposure, hepatic tumor induction by non-genotoxic chemicals, since there are currently no accurate and well-validated short-term tests to identify non-genotoxic hepatic tumorigens, thus necessitating an expensive 2-year rodent bioassay before a risk assessment can begin. The training set consists of hepatic gene expression data from 216 male Sprague-Dawley rats treated for 5 days with one of 76 structurally and mechanistically diverse nongenotoxic hepatocarcinogens and non-hepatocarcinogens. The validation set consists of 201 male Sprague-Dawley rats treated for 5 days with one of 68 structurally and mechanistically diverse non-genotoxic hepatocarcinogens and non- hepatocarcinogens. Gene expression data were generated using the Amersham Codelink Uniset Rat 1 Bioarray (GE HealthCare, Piscataway, NJ). The separation of the training set and validation set was based on the time when the microarray data were collected; i.e., microarrays processed earlier in the study were used as training and those processed later were used as validation. The NIEHS data set (endpoint C) was provided by the National Institute of Environmental Health Sciences (NIEHS) of the National Institutes of Health (Research Triangle Park, NC, USA). The study objective was to use microarray gene expression data acquired from the liver of rats exposed to hepatotoxicants to build classifiers for prediction of liver necrosis. The gene expression compendium data set was collected from 418 rats exposed to one of eight compounds (1,2- dichlorobenzene, 1,4-dichlorobenzene, bromobenzene, monocrotaline, N-nitrosomorpholine, thioacetamide, galactosamine, and diquat dibromide). All eight compounds were studied using standardized procedures, i.e. a common array platform (Affymetrix Rat 230 2.0 microarray), experimental procedures and data retrieving and analysis processes. Briefly, for each compound, four to six male, 12 week old F344 rats were exposed to a low dose, mid dose(s) and a high dose of the toxicant and sacrificed at 6, 24 and 48 hrs later. At necropsy, liver was harvested for RNA extraction, histopathology, and clinical chemistry assessments. The human breast cancer (BR) data set (endpoints D and E) was contributed by the University of Texas M. D. Anderson Cancer Center (MDACC, Houston, TX, USA). Gene expression data from 230 stage I-III breast cancers were generated from fine needle aspiration specimens of newly diagnosed breast cancers before any therapy. The biopsy specimens were collected sequentially during a prospective pharmacogenomic marker discovery study between 2000 and 2008. These specimens represent 70-90% pure neoplastic cells with minimal stromal contamination. Patients received 6 months of preoperative (neoadjuvant) chemotherapy including paclitaxel, 5-fluorouracil, cyclophosphamide and doxorubicin followed by surgical resection of the cancer. Response to preoperative chemotherapy was categorized as a pathological complete response (pCR = no residual invasive cancer in the breast or lymph nodes) or residual invasive cancer (RD), and used as endpoint D for prediction. Endpoint E is the clinical estrogen-receptor status as established by immunohistochemistry. RNA extraction and gene expression profiling were performed in multiple batches over time using Affymetrix U133A microarrays. Genomic analysis of a subset of this sequentially accrued patient population were reported previously. For each endpoint, the first 130 cases were used as a training set and the next 100 cases were used as an independent validation set. The multiple myeloma (MM) data set (endpoints F, G, H, and I) was contributed by the Myeloma Institute for Research and Therapy at the University of Arkansas for Medical Sciences (UAMS, Little Rock, AR, USA). Gene expression profiling of highly purified bone marrow plasma cells was performed in newly diagnosed patients with MM. The training set consisted of 340 cases enrolled on total therapy 2 (TT2) and the validation set comprised 214 patients enrolled in total therapy 3 (TT3). Plasma cells were enriched by anti-CD138 immunomagnetic bead selection of mononuclear cell fractions of bone marrow aspirates in a central laboratory. All samples applied to the microarray contained more than 85% plasma cells as determined by 2-color flow cytometry (CD38+ and CD45-/dim) performed after selection. Dichotomized overall survival (OS) and eventfree survival (EFS) were determined based on a two-year milestone cutoff. A gene expression model of high-risk multiple myeloma was developed and validated by the data provider and later on validated in three additional independent data sets. The neuroblastoma (NB) data set (endpoints J, K, L, and M) was contributed by the Children's Hospital of the University of Cologne, Germany. Tumor samples were checked by a pathologist prior to RNA isolation; only samples with =60% tumor content were utilized and total RNA was isolated from ~50mg of snap-frozen neuroblastoma tissue obtained before chemotherapeutic treatment. First, 502 pre-existing 11K Agilent dye-flipped, dual-color replicate profiles for 251 patients were provided. Of these, profiles of 246 neuroblastoma samples passed an independent MAQC-II quality assessment by majority decision and formed the MAQC-II training data set. Subsequently, 514 dyeflipped dual-color 11K replicate profiles for 256 independent neuroblastoma tumor samples were generated and profiles for 253 samples were selected to form the MAQC-II validation set. Of note, for one patient of the validation set, two different tumor samples were analyzed utilizing both versions of the 2x11K microarray (see below). All dual-color gene-expression of the MAQC-II training set were generated using a customized 2x11K neuroblastoma-related microarray. Furthermore, 20 patients of the MAQC-II validation set were also profiled utilizing this microarray. Dual-color profiles of the remaining patients of the MAQC-II validation set were performed using a slightly revised version of the 2x11K microarray. This version V2.0 of the array comprised 200 novel oligonucleotide probes whereas 100 oligonucleotide probes of the original design were removed due to consistent low expression values (near background) observed in the training set profiles. These minor modifications of the microarray design resulted in a total of 9,986 probes present on both versions of the 2x11K microarray. The experimental protocol did not differ between both sets and gene-expression profiles were performed as described. Furthermore, single-color geneexpression profiles were generated for 478/499 neuroblastoma samples of the MAQC-II dual-color training and validation sets (training set 244/246; validation set 234/253). For the remaining 21 samples no single-color data were available, due to either shortage of tumor material of these patients (n=15), poor experimental quality of the generated single-color profiles (n=5), or correlation of one single-color profile to two different dual-color profiles for the one patient profiled with both versions of the 2x11K microarrays (n=1). Single-color gene-expression profiles were generated using customized 4x44K oligonucleotide microarrays produced by Agilent Technologies (Palo Alto, CA, USA). These 4x44K microarrays included all probes represented by Agilent's Whole Human Genome Oligo Microarray and all probes of the version V2.0 of the 2x11K customized microarray that were not present in the former probe set. Labeling and hybridization was performed following the manufacturer's protocol as describeThe MAQC-II Project: A comprehensive study of common practices for the development and validation of microarray-based predictive models The second phase of the MicroArray Quality Control (MAQC-II) project evaluated common practices for developing and validating microarray-based models aimed at predicting toxicological and clinical endpoints. The purposes of the MAQC-II project were to survey approaches in genomic model development in an attempt to understand sources of variability in prediction performance, and to assess the influences of endpoint signal strength in data. Thirty-six teams developed classifiers for 13 diverse endpoints -- some easy, some difficult to predict, from six relatively large training data sets -- three preclinical (toxicogenomics) and three clinical. By providing the same data sets to many organizations for analysis, but not restricting their data analysis protocols (DAPs), the project made it possible to evaluate to what extent, if any, results depend on the team that performs the analysis. These analyses collectively produced >18,000 models that were challenged by independent and blinded validation sets generated for MAQC-II. The cross-validated performance estimates for models developed under good practices are predictive of the blinded validation performance. The achievable prediction performance is largely determined by the intrinsic predictability of the endpoint, and simple data analysis methods often perform as well as more complicated approaches. Multiple models of comparable performance can be developed for a given endpoint and the stability of gene lists correlates with endpoint predictability. Importantly, similar conclusions were reached when >12,000 new models were generated by swapping the original training and validation sets. Description of six data sets including 13 prediction endpoints: (Summarized in GSE16716_MAQC-II_Datasets_Overview.pdf attached as supplementary file. For more details, see the MAQC-II main paper and its references for individual dataset.) The MAQC-II predictive modeling was limited to binary classification problems; therefore, continuous endpoint values such as overall survival (OS) and event-free survival (EFS) times were dichotomized using a milestone cutoff of censor data. Prediction endpoints were chosen to span a wide range of prediction difficulty. Two endpoints, H (CPS1) and L (NEP_S), representing the gender of the patients, were used as positive control endpoints, since they are easily predictable by microarrays. Two other endpoints, I (CPS1) and M (NEP_R), representing randomly assigned class labels, were designed to serve as negative control endpoints, since they are not supposed to be predictable. Data analysis teams were not aware of the characteristics of endpoints H, I, L, and M until their swap prediction results had been submitted. If a data analysis protocol did not yield models to accurately predict endpoints H and L, or if a data analysis protocol claims to be able to yield models to accurately predict endpoints I and M, something must have gone wrong. The Hamner data set (endpoint A) was provided by The Hamner Institutes for Health Sciences (Research Triangle Park, NC, USA). The study objective was to apply microarray gene expression data from the lung of female B6C3F1 mice exposed to a 13-week treatment of chemicals to predict increased lung tumor incidence in the 2-year rodent cancer bioassays of the National Toxicology Program. If successful, the results may form the basis of a more efficient and economical approach for evaluating the carcinogenic activity of chemicals. Microarray analysis was performed using Affymetrix Mouse Genome 430 2.0 arrays on three to four mice per treatment group, and a total of 70 mice were analyzed and used as the MAQC-II's training set. Additional data from another set of 88 mice were collected later and provided as the MAQC-II's external validation set. The Iconix data set (endpoint B) was provided by Iconix Biosciences, Inc. (Mountain View, CA, USA). The study objective was to assess, upon short term exposure, hepatic tumor induction by non-genotoxic chemicals, since there are currently no accurate and well-validated short-term tests to identify non-genotoxic hepatic tumorigens, thus necessitating an expensive 2-year rodent bioassay before a risk assessment can begin. The training set consists of hepatic gene expression data from 216 male Sprague-Dawley rats treated for 5 days with one of 76 structurally and mechanistically diverse nongenotoxic hepatocarcinogens and non-hepatocarcinogens. The validation set consists of 201 male Sprague-Dawley rats treated for 5 days with one of 68 structurally and mechanistically diverse non-genotoxic hepatocarcinogens and non- hepatocarcinogens. Gene expression data were generated using the Amersham Codelink Uniset Rat 1 Bioarray (GE HealthCare, Piscataway, NJ). The separation of the training set and validation set was based on the time when the microarray data were collected; i.e., microarrays processed earlier in the study were used as training and those processed later were used as validation. The NIEHS data set (endpoint C) was provided by the National Institute of Environmental Health Sciences (NIEHS) of the National Institutes of Health (Research Triangle Park, NC, USA). The study objective was to use microarray gene expression data acquired from the liver of rats exposed to hepatotoxicants to build classifiers for prediction of liver necrosis. The gene expression compendium data set was collected from 418 rats exposed to one of eight compounds (1,2- dichlorobenzene, 1,4-dichlorobenzene, bromobenzene, monocrotaline, N-nitrosomorpholine, thioacetamide, galactosamine, and diquat dibromide). All eight compounds were studied using standardized procedures, i.e. a common array platform (Affymetrix Rat 230 2.0 microarray), experimental procedures and data retrieving and analysis processes. Briefly, for each compound, four to six male, 12 week old F344 rats were exposed to a low dose, mid dose(s) and a high dose of the toxicant and sacrificed at 6, 24 and 48 hrs later. At necropsy, liver was harvested for RNA extraction, histopathology, and clinical chemistry assessments. The human breast cancer (BR) data set (endpoints D and E) was contributed by the University of Texas M. D. Anderson Cancer Center (MDACC, Houston, TX, USA). Gene expression data from 230 stage I-III breast cancers were generated from fine needle aspiration specimens of newly diagnosed breast cancers before any therapy. The biopsy specimens were collected sequentially during a prospective pharmacogenomic marker discovery study between 2000 and 2008. These specimens represent 70-90% pure neoplastic cells with minimal stromal contamination. Patients received 6 months of preoperative (neoadjuvant) chemotherapy including paclitaxel, 5-fluorouracil, cyclophosphamide and doxorubicin followed by surgical resection of the cancer. Response to preoperative chemotherapy was categorized as a pathological complete response (pCR = no residual invasive cancer in the breast or lymph nodes) or residual invasive cancer (RD), and used as endpoint D for prediction. Endpoint E is the clinical estrogen-receptor status as established by immunohistochemistry. RNA extraction and gene expression profiling were performed in multiple batches over time using Affymetrix U133A microarrays. Genomic analysis of a subset of this sequentially accrued patient population were reported previously. For each endpoint, the first 130 cases were used as a training set and the next 100 cases were used as an independent validation set. The multiple myeloma (MM) data set (endpoints F, G, H, and I) was contributed by the Myeloma Institute for Research and Therapy at the University of Arkansas for Medical Sciences (UAMS, Little Rock, AR, USA). Gene expression profiling of highly purified bone marrow plasma cells was performed in newly diagnosed patients with MM. The training set consisted of 340 cases enrolled on total therapy 2 (TT2) and the validation set comprised 214 patients enrolled in total therapy 3 (TT3). Plasma cells were enriched by anti-CD138 immunomagnetic bead selection of mononuclear cell fractions of bone marrow aspirates in a central laboratory. All samples applied to the microarray contained more than 85% plasma cells as determined by 2-color flow cytometry (CD38+ and CD45-/dim) performed after selection. Dichotomized overall survival (OS) and eventfree survival (EFS) were determined based on a two-year milestone cutoff. A gene expression model of high-risk multiple myeloma was developed and validated by the data provider and later on validated in three additional independent data sets. The neuroblastoma (NB) data set (endpoints J, K, L, and M) was contributed by the Children's Hospital of the University of Cologne, Germany. Tumor samples were checked by a pathologist prior to RNA isolation; only samples with =60% tumor content were utilized and total RNA was isolated from ~50mg of snap-frozen neuroblastoma tissue obtained before chemotherapeutic treatment. First, 502 pre-existing 11K Agilent dye-flipped, dual-color replicate profiles for 251 patients were provided. Of these, profiles of 246 neuroblastoma samples passed an independent MAQC-II quality assessment by majority decision and formed the MAQC-II training data set. Subsequently, 514 dyeflipped dual-color 11K replicate profiles for 256 independent neuroblastoma tumor samples were generated and profiles for 253 samples were selected to form the MAQC-II validation set. Of note, for one patient of the validation set, two different tumor samples were analyzed utilizing both versions of the 2x11K microarray (see below). All dual-color gene-expression of the MAQC-II training set were generated using a customized 2x11K neuroblastoma-related microarray. Furthermore, 20 patients of the MAQC-II validation set were also profiled utilizing this microarray. Dual-color profiles of the remaining patients of the MAQC-II validation set were performed using a slightly revised version of the 2x11K microarray. This version V2.0 of the array comprised 200 novel oligonucleotide probes whereas 100 oligonucleotide probes of the original design were removed due to consistent low expression values (near background) observed in the training set profiles. These minor modifications of the microarray design resulted in a total of 9,986 probes present on both versions of the 2x11K microarray. The experimental protocol did not differ between both sets and gene-expression profiles were performed as described. Furthermore, single-color geneexpression profiles were generated for 478/499 neuroblastoma samples of the MAQC-II dual-color training and validation sets (training set 244/246; validation set 234/253). For the remaining 21 samples no single-color data were available, due to either shortage of tumor material of these patients (n=15), poor experimental quality of the generated single-color profiles (n=5), or correlation of one single-color profile to two different dual-color profiles for the one patient profiled with both versions of the 2x11K microarrays (n=1). Single-color gene-expression profiles were generated using customized 4x44K oligonucleotide microarrays produced by Agilent Technologies (Palo Alto, CA, USA). These 4x44K microarrays included all probes represented by Agilent's Whole Human Genome Oligo Microarray and all probes of the version V2.0 of the 2x11K customized microarray that were not present in the former probe set. Labeling and hybridization was performed following the manufacturer's protocol as described. This SuperSeries is composed of the SubSeries listed below.d. This SuperSeries is composed of the SubSeries listed below.This series represents the data set from the paper <u>Assembly of microarrays for genome-wide measurement of DNA copy number</u> appearing in Nat Genet 2001 Nov;29(3):263-4 and it's web supplement. Specimen procurement, labeling, hybridization and imaging protocols from Nat Genet 2001 Nov;29(3):263-4 web supplement: <b>Specimens.</b> We obtained GM03563, GM00143, GM05296, GM07408, GM01750, GM03134, GM13330, GM03576, GM01535, GM07081, GM02948, GM04435, GM10315, GM13031 and GM01524 cell strains from the NIGMS Human Genetics Cell Repository (Coriell Institute for Medical Research), and cultured them under the recommended conditions. We cultured cell lines BT474, HCT116, T47D, MPE600, COLO320, SW837, MDA-MB-231, MDA-MB-453 and HT29 under the recommended conditions. To isolate DNA from confluent cell cultures, we collected cells from T75 flasks and then incubated them overnight at 55 C in a 3 ml solution containing 0.01 M Tris, pH 7.5, 0.001 M EDTA, pH 8, 0.5% SDS and 0.1 g/l proteinase K. We added 1 ml of saturated NaCl solution and 10 ml of ethanol to the cell lysate and collected the DNA by spooling. We removed excess liquid and then dissolved the DNA in 1 ml of H2O. We isolated DNA from breast tumor specimens as described previously. <b>DNA labeling.</b> We labeled DNA by nick translation or random priming. For nick translation, we used a 400 l reaction containing 10 g DNA, 50 mM Tris, pH 7.6, 5 mM MgCl2, 0.02 mM each dATP, dTTP and dGTP, 0.2 mM Cy3- or Cy5-dCTP (Amersham), 0.2 U DNA Polymerase I (Gibco BRL), and 30 l DNAseI/PolI enzyme mix (Gibco BRL). We incubated the reaction at 15 C for 60 min after which we stopped the reaction by incubation at 70 C for 10 min. We removed unincorporated nucleotides using a Sephadex G-50 spin column. We labeled genomic DNA by random priming in a 100 l reaction containing 0.003 - 0.6 g DNA, 1x random primers solution (BioPrime DNA Labeling System, Gibco BRL), 1 mM Tris, pH 7.6, 0.1 mM EDTA, 0.2 mM each of dATP, dTTP and dGTP, 0.1 mM dCTP, 0.4 mM Cy3 or Cy5-dCTP (Amersham) and 160 U Klenow fragment (BioPrime DNA Labeling System, Gibco BRL). We incubated the DNA with the random primers solution at 100 C for 10 min in a total volume of 84 l, prior to adding the other reagents and then incubated the 100 l reaction overnight at 37 C. We removed unincorporated nucleotides using a Sephadex G-50 column. <b>Hybridization.</b> We combined test and reference DNAs (~2 g of input genomic DNA for nick translation or ~0.6 g of input genomic DNA for random prime labeling) with Cot-1 DNA (80-100 g; Gibco BRL) and precipitated them with ethanol. We collected the precipitate by centrifugation and allowed the precipitate to dry in air for 10 min before re-dissolving it in a 50 l hybridization mixture containing 50% formamide, 2 x SSC, 10% dextran sulfate, 4% SDS and 500 g yeast tRNA, pH 7. We incubated the hybridization mixture at 70 C for 10-15 min to denature the DNA and subsequently continued incubation at 37 C for 60 min to allow blocking of repetitive sequences. We applied a ring of rubber cement closely around the array to form a well, into which we added 50 l of slide blocking solution containing 500 g salmon sperm DNA in 50% formamide, 2 x SSC, 10% dextran sulfate and 4% SDS, pH 7. We created an airtight hybridization chamber by placing a silicone gasket (PGC Scientific) around the array and rubber cement ring, placing a slide on top of it and clamping with binder clips. After a 30 min incubation at room temperature, we opened the chamber, removed approximately three-quarters of the blocking solution, added the denatured and re-annealed hybridization mixture and then re-sealed the chamber. We placed the arrays on a slowly rocking table (~1 rpm) at 37 C to allow hybridization to occur over 16-72 h. After hybridization, we rinsed off the excess hybridization fluid with PN buffer (PN: 0.1 M sodium phosphate, 0.1% nonidet P40, pH 8), then washed once in 50% formamide, 2 x SSC, pH 7 at 45 C for 15 min, and finally in PN buffer at room temperature for 15 min. After draining excess liquid from the arrays, we mounted them in a solution containing 90% glycerol, 10% PBS and 1 M DAPI, and then sealed them with a cover slip. We drained excess glycerol-DAPI solution onto a paper tissue. <b>Imaging and analysis.</b> We acquired 16 bit 1024x1024 pixel DAPI, Cy3 and Cy5 images using a custom built CCD camera system, although we have also used commercial laser scanners for this purpose. We used ?UCSF SPOT? software (A.N.J. manuscript submitted) to automatically segment the spots based on the DAPI images, perform local background correction and to calculate various measurement parameters, including log2ratios of the total integrated Cy3 and Cy5 intensities for each spot. We used a second custom program SPROC to associate clone identities and a mapping information file with each spot so that the data could be plotted relative to the position of the BACs on the September, 2000 freeze of the draft human genome sequence (http://genome.ucsc.edu). SPROC also implements a filtering procedure to reject data based on a number of criteria, including low reference/DAPI signal intensity and low correlation of the Cy3 and Cy5 intensities with a spot. The SPROC output consists of averaged ratios of the triplicate spots for each clone, standard deviations of the triplicates and plotting position for each clone on the array, as well as other clone information stored in the database, such as STS content (Tables A and B). We edited the data files to remove ratios on clones for which only one of the triplicates remained after SPROC analysis and/or the standard deviation of the log2ratios of the triplicates was > 0.2. Keywords: otherBackground: Breast cancer is a complex disease, encompassed by different clinically and molecularly stratified entities. In 2000, Perou and colleagues demonstrated that tumor phenotypic diversity correlates with differences in global gene expression patterns, which in turn reflect aspects of the biological behavior of the tumors. Gene expression profiling has distinguished sporadic breast tumor sub-classes with genetic, histopathological and clinical differences, however, little is known about the molecular classification of hereditary breast tumors. It has been recently suggested that most tumors arising in BRCA1 mutation carriers display a basal-like phenotype, with the percentages reported in different studies ranging from 44 to almost 100%. In the present study we used expression profiling to produce a molecular classification of BRCA1-associated breast tumors and applied an integrative approach to examine biological dependencies or differences. Methods and Findings: We used frozen tumor tissue from fourteen patients all of which harbored germline pathological mutations in BRCA1. Total RNA extraction, amplification and labeling with Cy5 were performed using standard protocols. Universal Human Reference RNA (Stratagene) was used as a reference and labeled with Cy3. Each pair of Cy3/Cy5 samples was hybridized onto the CNIO human OncoChip V2, which consists of a spotted microarray with 11,675 cDNA clones from the I.M.A.G.E. Consortium. Two channel ratios (Cy5/Cy3) for each spot were generated and quantified using GenePix Pro 5.1 (Axon Instruments, Inc., Union City, CA, USA). Data were normalized and filtered and multiple statistical analyses were performed. The data are deposited in the GEO database under the accession number GSE12350. A tissue microarray (TMA) containing an independent series of 15 BRCA1 tumors was used to validate some of the results obtained. We have described molecular signatures that define BRCA1 subclasses depending on the expression of the gene encoding for Estrogen Receptor, ESR1. Signatures were found to be molecularly associated with different biological processes and transcriptional regulatory programs. The signature of ESR1-positive tumors was mainly linked to cell proliferation-related processes and, therefore, regulated by the estrogen receptor, while the signature of ESR1-negative tumors was mainly linked to the immune response and possibly regulated by transcription factors of the REL/NFκB family. These signatures were then verified in an independent series of hereditary and sporadic breast tumors, which revealed a possible prognostic value for each subclass. Over-expression of immune response genes appears to be a common feature of ER-negative sporadic and hereditary breast cancer and is associated with good prognosis. Interestingly, the ESR1-negative tumors were sub-stratified into two groups presenting light differences in the magnitude of the expression of immune response transcripts and REL/NFκB transcription factors, and this could be dependent on the type of germline alteration. In addition, analysis of the human protein-protein interaction network provides the wiring diagram of critical molecular associations in BRCA1 tumorigenesis and identifies close relationships between the different signatures. Conclusions: In summary, in the present study we have established the gene expression profiling of a series of BRCA1 tumors and found that there is a further degree of heterogeneity beyond the main classification by the expression of ESR1 and the presence or absence of a basal-like phenotype. We have identified specific signatures for ESR1-positive and ESR1-negative BRCA1 tumors, the latter characterized by the enrichment of immune response and cell cycle genes, and have found that slight differences in the level of expression of the immune response stratify the ESR1-negative BRCA1 tumors into two additional sub-groups (A and B) with possible prognostic differences. NFkB could be a major driver responsible for the levels of both immune response and apoptotic genes in this group of tumors, which could in turn be related to the type of germline mutation in BRCA1. This study reveals the molecular complexity of BRCA1 breast tumors, which are found to display similarities to sporadic tumors, and suggests possible prognostic implications.ABSTRACT: Introduction: Basal-like and luminal breast cancers have distinct stromal-epithelial interactions, which likely play a role in progression to invasive cancer. However, little is known about how stromal-epithelial interactions evolve in benign and pre-invasive lesions.Methods: To study epithelial-stroma interactions in basal-like breast cancer progression, we cocultured reduction mammoplasty fibroblasts with the isogenic MCF10 series of cell lines (representing benign/normal, atypical hyperplasia, and ductal carcinoma in situ). We used gene expression microarrays to identify pathways induced by coculture in premalignant cells (MCF10DCIS) compared to normal and benign (MCF10A and MCF10AT1). Relevant pathways were then (1) targeted in vitro and effects on morphogenesis were evaluated and (2) evaluated in vivo for associations with basal-like subtype. Results: Our results show that premalignant MCF10DCIS cells express characteristic gene expression patterns of invasive basal-like microenvironments. Furthermore, while HGF secretion is upregulated (relative to normal, MCF10A levels) when fibroblasts are cocultured with either atypical (MCF10AT cells) or premalignant (MCF10DCIS) cells, only MCF10DCIS cells upregulate the HGF receptor, MET. In 3-dimensional cultures, upregulation of HGF/MET in MCF10DCIS cells induced morphological changes suggestive of invasive potential, and these changes were reversed by antibody-based blocking of HGF signaling. These results are relevant to in vivo progression because high expression of a novel MCF10DCIS-derived HGF signature was correlated with basal-like subtype among invasive cancers, with approximately 86% of basal-like cancers highly expressing the HGF signature. Conclusions: In this study we document coordinated and complementary changes in HGF secretion and MET expression in epithelium and stroma in pre-invasive lesions. These results suggest that targeting stroma-derived HGF signaling in early carcinogenesis may block progression of basal-like precursor lesions.Introduction: In breast cancers, the basal-like subtype has high levels of genomic instability relative to other breast cancer subtypes with many basal-like-specific regions of aberration. There is evidence that this genomic instability extends to smaller scale genomic aberrations as well, as shown by a previously described micro-event in the PTEN gene in the Basal-like SUM149 breast cancer cell line. Methods: We sought to identify if small regions of genomic change exist by using a high density, gene centric Comparative Genomic Hybridizations (CGH) array on both cell lines and primary tumors. A custom Agilent tiling array for CGH (244,000 probes, 200bp tiling resolution) was created to identify small regions of genomic change and was focused on previously identified basal-like-specific, and general cancer genes. Tumor genomic DNA from 94 patients and 2 breast cancer cell lines was labeled and hybridized to these arrays. Aberrations were called using SWITCHdna and the smallest 25% of SWITCHdna-defined genomic segments being called micro-aberrations (<64 contiguous probes, ~ <15kb). Results: Our data showed that primary tumor breast cancer genomes frequently contained areas of small-scale copy number gains and losses, termed micro-aberrations, which are undetectable using lower-density genome-wide platforms. The basal-like subtype exhibited the highest incidence of these events. These micro-aberrations sometimes altered expression of the involved gene as suggested by data from microarray and mRNA-seq studies. We confirmed the presence of the PTEN micro-amplification in SUM149 and by mRNA-seq showed that this resulted in loss of expression of all exons downstream of this event. Micro-aberrations disproportionately affected the 5’ regions of the affected genes, including the promoter region, and a high frequency of micro-aberrations was associated with poor survival outcomes. Conclusion: Using a high probe density, gene-centric aCGH microarray, we present evidence of small-scale genomic aberrations that contribute to gene inactivation, and thus, genomic instability and tumor formation through a mechanism not detected using conventional copy number analyses.Basal-like breast cancer (BBC) is an aggressive subtype of breast cancer that has no biologically targeted therapy. The interactions of BBCs with stromal cells are important determinants of tumor biology, with inflammatory cells playing well-recognized roles in cancer progression. Despite the fact that macrophage–BBC communication is bidirectional, important questions remain about how BBCs affect adjacent immune cells. This study investigated monocyte-to-macrophage differentiation and polarization and gene expression in response to coculture with basal like versus luminal breast cancer cells. Changes induced by coculture were compared with changes observed under classical differentiation and polarization conditions. Monocytes (THP-1 cells) exposed to BBC cells in coculture had altered gene expression with upregulation of both M1 and M2 macrophage markers. Two sets of M1 and M2 markers were selected from the PCR profiles and used for dual immunofluorescent staining of BBC versus luminal cocultured THP-1s, and cancer-adjacent, benign tissue sections from patients diagnosed with BBCs or luminal breast cancer, confirming the differential expression patterns. Relative to luminal breast cancers, BBCs also increased differentiation of monocytes to macrophages and stimulated macrophage migration. Consistent with these changes in cellular phenotype, a distinct pattern of cytokine secretion was evident in macrophage–BBC cocultures, including upregulation of NAP-2, osteoprotegerin, MIG, MCP-1, MCP-3, and interleukin (IL)-1b. Application of IL-1 receptor antagonist (IL-1RA) to cocultures attenuated BBC-induced macrophage migration. These data contribute to an understanding of the BBC-mediated activation of the stromal immune response, implicating specific cytokines that are differentially expressed in basal-like microenvironments and suggesting plausible targets for modulating immune responses to BBCs.Introduction: Overall survival of early-stage breast cancer (BC) patients is similar for those who undergo breast conserving therapy (BCT) and mastectomy, however, 10-15% of women undergoing BCT suffer ipsilateral breast tumor recurrence. The risk of recurrence may vary with age or breast cancer subtype. Understanding the gene expression of the cancer-adjacent tissue and/or stromal response to specific tumor subtypes is important for developing clinical strategies to reduce recurrence risk. Methods: We studied gene expression data in cancer-adjacent tissue from 158 BC patients. Complementary in vitro cocultures were used to study cell-cell communication between fibroblasts and specific breast cancer subtypes. Results: Our results suggest that intrinsic tumor subtypes are reflected in histologically normal cancer-adjacent tissue. Gene expression of cancer-adjacent tissues shows that triple negative (Claudin-low or Basal-like tumors) exhibit increased expression of genes involved in inflammation and immune response. While such changes could reflect distinct immune populations present in the microenvironment of different breast cancer subtypes, altered immune response gene expression was also observed in cocultures in the absence of immune cell infiltrates, emphasizing that these inflammatory mediators are secreted by breast-specific cells. In addition, while triple negative BCs are associated with upregulated immune response genes, Luminal breast cancers are more commonly associated with estrogen-response in adjacent tissues. Conclusions: Specific characteristics of BCs are reflected in the surrounding benign tissue. This commonality between tumor and surrounding tissue may underlie second primaries and local recurrences. Biomarkers derived from cancer-adjacent tissue may be helpful in defining personalized surgical strategies or in predicting recurrence risk.Purpose: Development of resistance to tamoxifen is an important clinical issue in the treatment of patients with breast cancer. Tamoxifen resistance may be the result of the acquisition of epigenetic regulation such as DNA methylation within breast cancer cells resulting in changed mRNA expression of genes being pivotal for estrogen dependent growth. Alternatively, tamoxifen resistance may be due to selection of preexisting resistant cells, which may exhibit cancer stem-like characteristics or a combination of the two mechanisms. Methods: To evaluate the contribution of these possible mechanisms to tamoxifen resistance, we applied modified DNA methylation-specific digital karyotyping (MMSDK) and digital gene expression (DGE) in combination with massively parallel sequencing to analyze a well-established tamoxifen resistant cell line model: MCF-7/S0.5 (tamoxifen sensitive parental cell line) and 4 high-dosage tamoxifen selected resistant offspring sublines (MCF-7/TAMR-1, MCF-7/TAMR-4, MCF-7/TAMR-7 and MCF-7/TAMR-8). MMSDK uses BssHII as mapping enzyme (DNA methylation sensitive enzyme). Both MMSDK and DGE use NlaIII and MmeI to produce 20-21 bp tag. The indexed single-end sequencing was performed by Illumina HiSeq 2000 in BGI-Shenzhen. A dynamic programming algorithm-FASTX-Toolkit implemented in Perl was used to trim the adaptor sequence. The trimmed tags were subjected to quality filtering, so that only tags with sequencing quality higher than 30 for more than 80% of the nucleotides were used for subsequent analysis. For MMSDK tag mapping, we generated a simulated reference library, i.e., BssHII reference library, by in silico enzyme digestion of the human genome (hg19, UCSC) regardless of the methylation state. This library was used as reference for subsequent mapping of the tags in the MMSDK analysis. In the DGE analysis, refMrna (hg19, UCSC) was used as reference for mapping cDNA tags. Subsequently, the Burrows–Wheeler Aligner (BWA) procedure for aligning the MMSDK and DGE tags to the simulated BssHII reference library and refMrna reference library, respectively, was applied. Results: MMSDK libraries using BssHII/NlaIII were generated from the parental tamoxifen sensitive subline MCF-7/S0.5 and the 4 TAMR cell lines: TAMR-1, TAMR-4, TAMR-7 and TAMR-8. The 5 indexed MMSDK libraries were sequenced in one lane and 1.38 Gb clean tag data for all 5 cell lines were obtained, with an average sequencing amount of ~270 Mb per library. On average, 59.5 % of the tags with mapping quality ≥ 20 were mapped back to the simulated BssHII/NlaIII reference library. DGE libraries were also generated from MCF-7/S0.5 and the 4 TAMR cell lines. The 5 indexed DGE libraries were sequenced in one lane and obtained 1.71 Gb clean tag data for all 5 cell lines with an average sequencing amount of ~340 Mb per library. On average, 40.8 % with mapping quality ≥ 20 were mapped back to the simulated NlaIII human transcriptome (refMrna reference library). Our present study demonstrates large differences in global gene expression and DNA methylation profiles between parental tamoxifen-sensitive cell line and 4 high-dosage tamoxifen treatment selected resistant sublines. The tamoxifen resistant cell lines exhibited globally higher methylation level than the parental cell line and an inverse relationship between gene expression and DNA methylation in the promoter regions were noticed. High expression of SOX2 and alterations of other SOX gene family members, E2F gene family members and RB-related pocket protein genes as well as highlighted stem cell pathways imply that cancer initiating cells/stem cells are involved in the resistance to tamoxifen.This data was generated by ENCODE. If you have questions about the data, contact the submitting laboratory directly (Richard Sandstrom mailto:sull@u.washington.edu). If you have questions about the Genome Browser track associated with this data, contact ENCODE (mailto:genome@soe.ucsc.edu). This track contains chromatin interaction data from the University of Washington ENCODE group generated using 5C (Chromatin Conformation Capture Carbon Copy). The 5C method is used here to define short and long-range range interactions between transcription start sites (TSS) and DNaseI hypersensitive sites (DHS) or other genomic features. The 5C method is summarized below. Transcription factors bind to promoter-associated proteins, bringing the associated DNA sequences in close proximity to each other. Cross linking the DNA and proteins immobilizes these interactions and thus maintains their close proximity. Cleavage of the sample with restriction endonuclease followed by ligation results in hybrid molecules where a fragment with a regulatory element is physically associated with a fragment containing a TSS. The interactions are then detected by oligonucleotide-dependent, ligation-mediated assays, where one set of primers is complementary to the end of fragments with a TSS and the second set of primers are complementary to fragments with a feature. Primers are designed to the forward strand of the feature and the reverse strand of the TSS so that ligation only occurs between TSS and feature, not between different features. Specific interactions are detected by massively parallel sequencing. The data in this track comprises two different experiment types focusing on targeted regions: Gene-targeted project: Analysis of DNase I hypersensitive sites reveals many genes where there are multiple sites restricted to the cell type where a protein is observed to be expressed. These sites potentially identify regulatory sites for the gene. This set of experiments attempts to observe interactions between these DHS sites and transcription starts in 25 regions selected based on genes expressed in GM06990 (B-lymphocyte), BJ (foreskin fibroblast), HepG2 (liver cancer cell line), or SK-N-SH_RA (neuroblastoma cell line, SKNSH, differentiated with retinoic acid). Myc project: Genome wide association studies have identified SNPs linked to prostate, colon, and breast cancer in the gene desert region upstream of the myc gene. 5C of HindIII fragments interacting with those containing refSeq txStarts in this region were performed in 5 cell types: GM12878 (B-lymphocyte), CaCo2 (colon cancer cell line), LNCaP (prostate cancer cell line), MCF7 (breast cancer cell line), and K562 (erythroleukemia cell line). For data usage terms and conditions, please refer to http://www.genome.gov/27528022 and http://www.genome.gov/Pages/Research/ENCODE/ENCODEDataReleasePolicyFinal2008.pdf";
		System.out.println(str);
		str = str.replaceAll("[^\\p{ASCII}]", "");
		System.out.println("After removing non ASCII chars:");
		System.out.println(str);

		writeFile(dir + "allsummarys-non.txt", str);
	}

	public static void writeFile(String filepath, String text) {
		try {
			// create a temporary file
			File logFile = new File(filepath);

			BufferedWriter writer = new BufferedWriter(new FileWriter(logFile));
			writer.write(text);

			// Close writer
			writer.close();
		} catch (Exception e) {
			e.printStackTrace();
		}
	}
}